PWGHF/HFC/TableProducer/correlatorDsHadrons.cxx

```diff
@@ -689,32 +692,77 @@ struct HfCorrelatorDsHadrons {
   }
   PROCESS_SWITCH(HfCorrelatorDsHadrons, processMcGen, "Process MC Gen mode", false);
 
-  void processDerivedDataDs(SelCollisionsWithDs::iterator const& collision,
+  int hfcReducedCollisionIndex = 0;
```

Comment:
Don't declare data members between methods. It makes the code structure hard to navigate.

Reply:
Hi @vkucera, if the process function is called for each DF analyzed in this way the index of the collisions restarts the counting for each collision every DF. I'm just wandering if in this case there will be candidates from different collisions but with the same index.

Reply:
I don't understand the relevance of your comment. This line declares a data member with an initial value of 0. It has nothing to do with resetting the value during processing. The variable is initialised to 0 when an object of type `HfCorrelatorDsHadrons` is created. If you want to reset its value for each call of a member method, you have to do it inside that method.

Reply:
I tested the code locally by inserting the data member declaration inside the process function and attached some plots to illustrate the results. As shown, 6 collisions are saved, but the indices of the Ds candidates are all set to 0. I was wondering whether, in this case, all the candidates are pointing to the first collision in the table, or if each candidate correctly points to its respective collision.
![Schermata del 2024-12-17 10-20-09](https://github.com/user-attachments/assets/b3e12b04-35f9-41f5-9153-ebf21ba9ee42)
![Schermata del 2024-12-17 10-20-26](https://github.com/user-attachments/assets/5ebeb16d-054c-410f-a36b-42633a4e1c6c)
![Schermata del 2024-12-17 10-20-44](https://github.com/user-attachments/assets/b366d4d8-90e4-4e63-9d81-2d80a88d79b9)

On the other hand, when I declare the data member outside the process function, I obtain the second set of plots, where 21 candidates are reconstructed over 20 collisions, and the candidate indices span from 0 to 19. To me, this suggests that each candidate is correctly associated with the collision it belongs to.
![Schermata del 2024-12-17 10-21-01](https://github.com/user-attachments/assets/02ed8334-75ea-4647-bc5d-711190884c1e)
![Schermata del 2024-12-17 10-21-09](https://github.com/user-attachments/assets/72dfa70e-92ad-44b3-b703-3a41f7fc6528)

Reply:
I'm afraid there have been a few misunderstandings.
- I did not ask you to declare the member inside the process function which would make it a local variable. I asked you not to declare it between methods, referring to the order of declaration of `struct `members.
- Your process function has an explicit loop over collisions. And you increment `hfcReducedCollisionIndex` at the end of each iteration. If the result of your local test was the same value for each collision, I suspect that you wrongly declared it inside the loop instead of outside.
- It is common to increment counters with the `++` operator in C++.